### PR TITLE
Topbar mobile menu header

### PR DESCRIPTION
**Pull request by kpuputti**
Avatar (that links to the user profile) + "Post a new listing" button in the top of the mobile menu.

screen shot 2016-06-30 at 12 53 01

[Interactive example](https://sharetribe.github.io/sharetribe/topbar-mobile-menu-header/?selectedKind=Top%20bar&selectedStory=Basic%20state) in the Styleguide.

### CODE REVIEW

File: client/app/components/composites/MenuMobile/MenuMobile.css [hunk 1]
```diff
@@ -60,7 +60,7 @@
   overflow-x: hidden;
   overflow-y: auto;
   will-change: transform;
-  transform: translateX(-288px);
+  transform: translateX(-300px);
```
Comment (mporkola) [inline client/app/components/composites/MenuMobile/MenuMobile.css]: maybe a variable for these two `300px`s?

File: client/app/components/composites/MenuMobile/OffScreenMenu.js [hunk 2]
```diff
@@ -5,21 +5,26 @@ import css from './MenuMobile.css';
...
 import AddNewListingButton from '../../elements/AddNewListingButton/AddNewListingButton';
+import LoginLinks from '../../composites/LoginLinks/LoginLinks';
 
 class OffScreenMenu extends Component {
 
   render() {
     const isOpenClass = this.props.isOpen ? css.offScreenMenuOpen : '';
 
+    const header = this.props.avatar ? [
+      this.props.avatar ? r(Avatar, this.props.avatar) : null,
```
Comment (mporkola) [inline client/app/components/composites/MenuMobile/OffScreenMenu.js]: This check is probably redundant, if it's also checked before entering the branch. I think detecting login state by presence of `avatar` might not work though, I'd also do the backend changes in this branch or soon after merge.

File: client/app/components/sections/Topbar/Topbar.js [hunk 3]
```diff
@@ -86,30 +86,6 @@ const profileLinks = function profileLinks(username, router, location, customCol
...
-      customColor,
-    }, t('web.topbar.login')),
-  ]);
-
-LoginLinks.propTypes = {
-  loginUrl: PropTypes.string.isRequired,
-  signupUrl: PropTypes.string.isRequired,
-  customColor: PropTypes.string,
-};
-
```
Comment (mporkola) [inline client/app/components/sections/Topbar/Topbar.js]: <3 cleanup

File: client/app/components/sections/Topbar/Topbar.js [hunk 4]
```diff
@@ -166,10 +142,13 @@ class Topbar extends Component {
     const profileRoute = this.props.routes && this.props.routes.person_path && loggedInUsername ?
             this.props.routes.person_path(loggedInUsername) :
             null;
-    const mobileMenuAvatarProps = this.props.avatarDropdown ?
+    const mobileMenuAvatarProps = this.props.avatarDropdown && loggedInUsername ?
```
Comment (mporkola) [inline client/app/components/sections/Topbar/Topbar.js]: ok, this probably answers my earlier question about login state

**Comment by mporkola** [comment]
Couple of simple comments, good to go besides them

File: client/app/components/composites/MenuMobile/MenuMobile.css (hunk 1 above)
Comment (kpuputti) [inline client/app/components/composites/MenuMobile/MenuMobile.css]: Good point, got caught already once when I only changed the other one and things didn't look right.

File: client/app/components/composites/MenuMobile/OffScreenMenu.js (hunk 2 above)
Comment (kpuputti) [inline client/app/components/composites/MenuMobile/OffScreenMenu.js]: As you mention in the latter comment, yes, the earlier code checks for avatarDropdownProps, but even if there are those props defined, this component gets a null in the avatar prop when the user is logged out.

**Pull request merged**